server/middleware/permissionsMiddleware.ts

```diff
@@ -0,0 +1,40 @@
+import { RequestHandler } from 'express'
+import { services } from '../services'
+import logger from '../../logger'
+import AuthorisedRoles from '../authentication/authorisedRoles'
+import Permissions from '../authentication/permissions'
+
+export default function populateUserPermissions(): RequestHandler {
+  const { keyworkerApiService } = services()
+
+  return async (req, res, next) => {
+    try {
+      const userIsKeyworker =
+        res.locals.user.userRoles.includes(AuthorisedRoles.KEYWORKER_MONITOR) ||
+        (await keyworkerApiService.isKeyworker(
+          req,
+          res.locals.user.activeCaseLoad!.caseLoadId!,
+          res.locals.user.username,
+        ))
+
+      res.locals.user.permissions = []
```

Comment:
Rather than an array which requires the `res.locals.user.permissions.includes(Permissions.View)` check further down, how about permissions = { view: true/false} etc?

I believe the view has direct access to res.locals so could be checked from there rather than passing to the view